### PR TITLE
move http transport into own module

### PR DESCRIPTION
**Pull request by chaudum**
* the netty server transport now lives in its own module
* services can add handlers to the http channel pipeline
* unused ssl/https was removed
* removed manual starting of HttpServer since it is not necessary any more with ES5
* inject classes directly instead of using injector

### CODE REVIEW

File: http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    @Override
+    public ChannelPipelineFactory configureServerChannelPipelineFactory() {
+        return new CrateHttpChannelPipelineFactory(this, detailedErrorsEnabled, threadPool);
+    }
+
+    public void addBefore(ChannelPipelineItem item) {
+//        // override item with same name because the pipeline does only accept unique handler names
+//        addBeforeList.removeIf(o -> o.name.equals(item.name));
```
Comment (seut) [inline http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java]: remove?

File: http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        addBeforeList.add(item);
+    }
+
+    public void addAfter(ChannelPipelineItem item) {
+//        // override item with same name because the pipeline does only accept unique handler names
+//        addAfterList.removeIf(o -> o.name.equals(item.name));
```
Comment (seut) [inline http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java]: here also?

File: http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java (hunk 1 above)
Comment (chaudum) [inline http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java]: yepp, forgot about it

**Comment by chaudum** [comment]
@seut move the starting and stopping of the http server into a new HttpTransportService

**Comment by chaudum** [comment]
I will leave https://github.com/crate/crate/pull/5540/commits/f3489c4b9ddec449be3fe14d88b1b70478fd85ec as separate commit

**Comment by seut** [comment]
pls leave the http server start/stop removal as a separate commit as well

**Pull request merged**